3-UsingAPIs/Week2/homework/ex1-programmerFun/index.js

```diff
@@ -14,30 +14,41 @@
 6. Test error handling, for instance, by temporarily changing the `.sh` in the 
    url with `.shx`. There is no server at the modified url, therefore this 
    should result in a network (DNS) error.
+   
 ------------------------------------------------------------------------------*/
 function requestData(url) {
-  // TODO return a promise using `fetch()`
+  return fetch(url).then((response) => {
+    if (response.ok) {
+      return response.json();
+    }
+    throw new Error('Request failed');
+  });
 }
 
 function renderImage(data) {
-  // TODO render the image to the DOM
-  console.log(data);
+  const imgUrl = data.img;
+  const imgAlt = data.alt;
+  const body = document.querySelector('body');
+  const img = document.createElement('img');
+  img.setAttribute('src', imgUrl);
+  img.setAttribute('alt', imgAlt);
+  body.appendChild(img);
 }
 
 function renderError(error) {
-  // TODO render the error to the DOM
-  console.log(error);
+  const body = document.querySelector('body');
+  const h1 = document.createElement('h1');
+  h1.textContent = error;
+  body.appendChild(h1);
```

Comment:
This works fine, but you also use this and remove line 39.

```suggestion
  document.body.appendChild(h1);
```